### PR TITLE
Scratch/colorize output strings

### PR DESCRIPTION
**Pull request by bkolligs**
Adds macros that will color a string that is published to the terminal. 

## Description
Colors offered: 
* Red
* Green
* Yellow
* Blue
* Magenta
* Cyan
There are normal font and bold fonts available for each color. 

This change will assist sorting through visual noise during cFS startup. 

## How has this been tested?
See screen shot for successful test. 

## Screenshots (if appropriate):
This screenshot displays each color
image

And this screenshot displays why this is useful: 
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read and followed the `CONTRIBUTING.md` file
- [x] I have applied the .clang-format file to my changes
- [x] I have commented my code using Doxygen style comments (see Contributing MD for examples), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design documentation 
- [ ] I have updated any relevant parameters in the internal ICD
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this is a bug-fix, I have added tests that can catch similar bugs in the future
- [ ] New and existing unit tests pass locally with my changes
- [ ] For cfe apps I have reserved an empty address for my msgs_ids and perf_ids from the [tracking list](https://docs.google.com/spreadsheets/d/1yRUhNTRBOiGLswlsWftNaYlwGKDeNQ_fCnDYFrc_7fU/edit#gid=772812381) and checked that it does not clash with other apps
- [ ] For new added message types, there is a timestamp included via either an explicit timestamp variable in the struct or buried in the a tlmHeader.

### CODE REVIEW

File: fsw/public_inc/moonranger_output_strings.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#define _moonranger_output_strings_h_
+
+#define ESCAPE    "\033["
+#define TERMINATE "m"
+/* Reset color back to normal with ANSI color codes */
+#define RESET "0"    TERMINATE
+#define END   ESCAPE RESET
+
+/* Define ANSI color codes for strings */
+#define RED            "31" TERMINATE
```
Comment (kangkelvin) [inline fsw/public_inc/moonranger_output_strings.h:24]: We should have a standard of what the color means.
E.g.
RED - critical error
YELLOW - warning
GREEN - smth went well, but don't spam too much green
NO COLOR - standard debug

Comment (bkolligs) [reply fsw/public_inc/moonranger_output_strings.h:24]: Agreed

File: fsw/public_inc/moonranger_output_strings.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#define BLUE_BOLD_STRING(string)    ESCAPE BLUE_BOLD    string END
+#define MAGENTA_STRING(string)      ESCAPE MAGENTA      string END
+#define MAGENTA_BOLD_STRING(string) ESCAPE MAGENTA_BOLD string END
+#define CYAN_STRING(string)         ESCAPE CYAN         string END
+#define CYAN_BOLD_STRING(string)    ESCAPE CYAN_BOLD    string END
+
+/* Define macros to represent the type of information we are presenting */
+#define ERROR_INFO(string)   RED_STRING(string)
+#define WARNING_INFO(string) YELLOW_STRING(string)
+#define SUCCESS_INFO(string) GREEN_STRING(string)
```
Comment (bkolligs) [inline fsw/public_inc/moonranger_output_strings.h:55]: We can use these macros preferentially

**Comment by kangkelvin** [comment]
@bkolligs need your opinion on this. Do you think the `#define` is unique enough? Because there is a danger with using `#define` with very common words as it might change a piece of code it is not meant to change. You have `#define RED` and `RED` might be found somewhere else in our flight repo

A workaround would be for example adding an `MR_` or `MOON_` or `COLOR_` to the start of all the `#define` to make them more unique and less likely to clash

**Comment by bkolligs** [comment]
> @bkolligs need your opinion on this. Do you think the `#define` is unique enough? Because there is a danger with using `#define` with very common words as it might change a piece of code it is not meant to change. You have `#define RED` or `#define END` and `RED`/`END` might be found somewhere else in our flight repo not meant for coloring
> 
> A workaround would be for example adding an `MR_` or `MOON_` or `COLOR_` to the start of all the `#define` to make them more unique and less likely to clash

I think this is a valid concern. However during my testing, there was no name clashes. That said, I think adding COLOR_ infront of the specific colors may increase readability when these show up in people's intellisense.

**Comment by bkolligs** [comment]
> @bkolligs need your opinion on this. Do you think the `#define` is unique enough? Because there is a danger with using `#define` with very common words as it might change a piece of code it is not meant to change. You have `#define RED` or `#define END` and `RED`/`END` might be found somewhere else in our flight repo not meant for coloring
> 
> A workaround would be for example adding an `MR_` or `MOON_` or `COLOR_` to the start of all the `#define` to make them more unique and less likely to clash

I created more unique names for everything, and performed a search throughout the repo to verify they didn't clash.

**Review by tenzincrouch — APPROVED** [review]
LGTM.

Thanks for putting this together.

**Comment by tenzincrouch** [comment]
@bkolligs - can you post an update in software explaining how to use this and also add it to the wiki?

**Pull request merged**